interview-150/274.ts

```diff
@@ -0,0 +1,11 @@
+function hIndex(citations: number[]): number {
+    const sortedArr = citations.sort((a, b) => b - a)
+
+    for (let i = 0; i < sortedArr.length; i++) {
+        if (i >= sortedArr[i]) {
+            return i
+        }
```

Comment:
It would be helpful to add a comment here to explain _why_ this solves the puzzle (generally speaking, this is a key thing to start adding to your code).